app/components/UserAttendance/AttendanceModal.tsx

```diff
@@ -48,98 +48,99 @@ const Tab = ({ name, index, activePoolIndex, togglePool }: TabProps) => (
   </Button>
 );
 
-type State = {
-  pools: Pool[];
-  filter: string;
-};
+const AttendanceModal = ({
+  title = 'Status',
+  pools,
+  togglePool,
+  selectedPool,
+  isMeeting,
+}: Props) => {
+  const [amendedPools, setAmendedPools] = useState<Pool[]>([]);
+  const [registrations, setRegistrations] = useState<Registration[]>([]);
+  const [filter, setFilter] = useState<string>('');
 
-class AttendanceModal extends Component<Props, State> {
-  state = {
-    pools: [],
-    filter: '',
-  };
-  static defaultProps = {
-    title: 'Status',
-  };
+  useEffect(() => {
+    generateAmendedPools(pools);
+  }, [pools]);
 
-  UNSAFE_componentWillMount() {
-    this.generateAmendedPools(this.props.pools, this.props.allRegistrations);
-  }
+  const generateAmendedPools = (pools: Pool[]) => {
+    if (pools.length === 1) return setAmendedPools(pools);
 
-  generateAmendedPools = (
-    pools: Array<Pool>,
-    allRegistrations?: Registration[]
-  ) => {
-    if (pools.length === 1)
-      return this.setState({
-        pools,
-      });
-    const registrations =
-      allRegistrations || flatMap(pools, (pool) => pool.registrations);
```

Comment:
This is what fixed the problem. `allRegistrations` did not include waiting list registrations.